cli/tests/pcluster/utils.py

```diff
@@ -65,6 +65,38 @@ def get_asset_content_with_resource_name(assets, resource_name):
     return None
 
 
+def get_resource_from_assets(assets, resource_name):
+    """Return the Resource dictionary containing a specific resource."""
+    asset = get_asset_content_with_resource_name(assets, resource_name)
+    if asset:
+        return asset["Resources"][resource_name]
+
+
+# It expects as input ['UserData']['Fn::Base64']['Fn::Sub'] that is a list of two elements
+# 1: contains a dictionary with the key-values to be replaced by Fn::Sub
+# 0: contains the file (as str) where those values will be replaced
+def render_user_data(user_data):
+    """Render the UserData as CF will do."""
+    content = user_data[0]
+    values = user_data[1]
+
+    for value in values:
+        content = content.replace("${%s}" % value, str(values[value]))
+
+    return content
+
+
+def validate_dna_json_fields(user_data, fields):
```

Comment:
Can you please add a comment (or typing) to describe what the `user_data` variable should contain?

Reply:
Will do

Reply:
Done